tests/mir-opt/inline/unwrap_unchecked.unwrap_unchecked.PreCodegen.after.panic-abort.mir

```diff
@@ -26,23 +24,16 @@ fn unwrap_unchecked(_1: Option<T>) -> T {
     bb0: {
         StorageLive(_2);
         _2 = discriminant(_1);
-        switchInt(move _2) -> [0: bb1, 1: bb2, otherwise: bb3];
+        switchInt(move _2) -> [0: bb2, 1: bb1, otherwise: bb2];
     }
 
     bb1: {
-        StorageLive(_3);
-        _3 = UbChecks();
-        assume(_3);
```

Comment:
That's very odd, why was there an `assume` here saying that we *must* be having UbChecks...?

Reply:
Ah, I guess there used to be an "if !UbChecks then unreachable" and that already got eliminated and turned into an `assume`? I didn't know we are doing transforms like that, interesting. Given past issues with `assume` confusing LLVM I am a bit surprised that this is being done, but I assume people had their reasons. :)

Reply:
> Given past issues with `assume` confusing LLVM I am a bit surprised that this is being done, but I assume people had their reasons. :)

The fact of the range of the value is lost if unreachable_prop doesn't add `assume`. This could potentially prevent some optimization based on that fact.

Reply:
Is nice to see you were also surprised by this MIR. I spent a lot of time squinting at it before.